catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -354,6 +364,40 @@ public String getOrganization() {
         return null;
     }
 
+    @Override
+    public Optional<MetacardValidationReport> validateMetacard(Metacard metacard) {
+
+        MetacardValidationReportImpl report = null;
+        try {
+            validate(metacard);
+        } catch (SchematronValidationException e) {
```

Comment:
Instead of controlling flow using exceptions, I would suggest splitting `validate` into two methods.  One would have the current signature but the other would extract most of the code and return a `SchematronReport`.  This `validateMetacard` method would then just use the `SchematronReport` instead of parsing the thrown `SchematronValidationException`.